code/modules/clothing/masks/chemmask.dm

```diff
@@ -0,0 +1,335 @@
+//inject it into the wearer. The amount taken from the primary tank is determined by tank_injection_rate, which defaults
+//to 10, and can be altered by the user with a verb. The amount taken from the primary tank is altered slightly per-reagent,
+//based on that reagent's custom_metabolism value, in order to avoid taking too much of a reagent just because that reagent
+//happens to metabolize more slowly than others.
+
+//Additionally, the chemical pack has an auxiliary chamber which can store any /obj/item/weapon/reagent_containers/glass item.
+//While capped at the largest beaker size, currently 300u, this chamber offers more control over the way in which reagents
+//are administered from it. It defaults to a threshold-based system, in which it injects 10u of creatine when the user falls
+//below 10u of creatine in their body. However, the user can alter the settings and change it to a time-based system, change
+//the time interval, the amount injected, and the reagent monitored by the threshold system, through verbs.
+
+//By default, the primary tank is connected to the mask, and the auxiliary beaker is disconnected. The user may toggle these
+//connections using verbs. When disconnected, the mask will not take any additional reagents from the primary tank and/or the
+//auxiliary beaker. Cycling these verbs will not force injections, the tank is still only capable of one injection per 97 seconds,
+//and the beaker has no need to be forced to inject since the user can alter its time interval directly.
+
+/obj/item/clothing/mask/chemmask
+	desc = "A rather sinister mask designed for connection to a chemical pack, providing the pack's safeties are disabled."
+	name = "chemical mask"
+	icon_state = "chemmask0"
+	flags = FPRINT | MASKINTERNALS
+	w_class = 2
+	var/power = 0
+	gas_transfer_coefficient = 0.01
+	permeability_coefficient = 0.01
+	var/tank_injection_rates = list(5,10,15,20,25,30,35,40,45,50)
+	var/tank_injection_rate = 10
+	var/tank_has_injected = 0
+	var/beaker_injection_methods = list("Time-based","Threshold-based")
+	var/beaker_injection_method = "Threshold-based"
+	var/injection_method_chosen = 0
+	var/beaker_time_intervals = list(1,5,10,15,20,25,30,40,50,60,70,80,90,97,100,120,180,240,300)
+	var/beaker_time_interval = 970
+	var/beaker_injection_rates = list(5,10,15,20,25,30,35,40,45,50)
+	var/beaker_injection_rate = 10
+	var/beaker_thresholds = list(1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,20,30,40,50)
+	var/beaker_threshold = 10
+	var/beaker_threshold_reagents = list("creatine","blood","anti_toxin","bicaridine","kelotane","dermaline","dexalinp","iron","hyperzine","doctorsdelight","inaprovaline","space_drugs","tramadol","oxycodone","dexalin","leporazine","tricordrazine","hyronalin","alkysine","imidazoline","peridaxon","peptobismol")
+	var/beaker_threshold_reagent = "creatine"
+	var/empty = 0
+	var/beakerempty = 0
+	var/hasbeaker = 0

```

Comment:
`beaker` is a variable of the chempack. I'm using a bool here because otherwise I'd need to do
```
var/obj/item/weapon/reagent_containers/chempack/P = user.back
	if (P.beaker)
```
Anywhere I wanted to see if there was a beaker instead of just checking the bool.

Or is there a better way to check the value of another object's var?